### PR TITLE
Support updating of the CLI *from* the CLI

### PR DESCRIPTION
**Pull request by jbaxleyiii**
> I'm going to use the PR for testing but will open another with the appropriate parts split out for review

This (wip) PR aims to do two things:
- `ap update` which checks for the latest build and installs over the current binary if its newer
- a background process which checks for updates to let people know a new build is ready

auto_update

### CODE REVIEW

File: cli/src/commands/update.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        download.show_progress(true);
+
+        download.download_to(&mut tmp_archive).unwrap();
+        debug!(
+            "Downloaded newest release to {}. Starting to extract tarball",
+            tmp_archive_path.to_string_lossy()
+        );
+
+        Extract::from_source(&tmp_archive_path)
+            .extract_file(&tmp_dir, "dist/ap")
```
Comment (jbaxleyiii) [inline cli/src/commands/update.rs:131]: ah! I can get the `ap` here from the name of the file 🎉

File: graphql-parser/tests/tests.rs [hunk 2]
```diff
@@ -15,52 +15,52 @@ use graphql_parser::parse_schema;
 
 
 
-test!(interface, include_str!("interface.graphql"), include_str!("interface.graphql"));
```
Comment (jbaxleyiii) [inline graphql-parser/tests/tests.rs:18]: hmmm, I think my fmt did this. I'll remove this change

**Comment by designatednerd** [comment]
Obligatory

 
yo dawg I heard you like CLI updates so I made a CLI to update your CLI

**Comment by jbaxleyiii** [comment]
Closed in favor of three other PRs #44, #45, #46

**Pull request closed without merging**